### PR TITLE
Remove tests from official builds

### PR DESCRIPTION
**Pull request by dougbu**
- #22787

nit: _add_ dependency on Windows ARM64 build when publishing to the BAR
    - not a major problem because this particular build rarely if ever fails
    - the existence of the correct manifests is much more important

### CODE REVIEW

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -675,18 +676,15 @@ stages:
         dependsOn:
           - Windows_build
           - Windows_arm_build
+          - Windows_64_build
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:679]: /fyi @JunTaoLuo as I mentioned in the description, this was not a big deal but is something to keep in mind if / when we add more build jobs e.g. if we split the Linux_x64 build up

Comment (wtgodbe) [reply .azure/pipelines/ci.yml:679]: Why did we name that job `Windows_64`? Shouldn't it be `Windows_Arm64`?

**Review by wtgodbe — APPROVED** [review]

**Comment by mmitche** [comment]
You're my hero @dougbu!

**Review by mmitche — APPROVED** [review]

File: .azure/pipelines/ci.yml (hunk 1 above)
Comment (dougbu) [reply .azure/pipelines/ci.yml]: Fixed that @wtgodbe

File: .azure/pipelines/ci.yml [hunk 2]
```diff
@@ -675,18 +676,15 @@ stages:
         dependsOn:
           - Windows_build
           - Windows_arm_build
+          - Windows_arm64_build
```
Comment (dougbu) [inline .azure/pipelines/ci.yml:679]: /fyi for @JunTaoLuo stands 😺

**Comment by dougbu** [comment]
Just remembered to kick off an internal build to confirm this works as expected. Things in https://dev.azure.com/dnceng/internal/_build/results?buildId=692395 look good:
image

(I left the source build alone because it rarely fails and might start doing real stuff in official builds soon-ish.)

**Comment by dougbu** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=692395 wasn't perfect but the problem was macOS-specific. Ignoring that glitch&hellip;

File: .azure/pipelines/ci.yml (hunk 2 above)
Comment (JunTaoLuo) [reply .azure/pipelines/ci.yml:679]: 👍

**Comment by JunTaoLuo** [comment]
🎉

**Pull request merged**